src/CQRS/LeanCode.CQRS.MassTransitRelay/LockProviders/CustomSqlLockStatementProvider.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections.Concurrent;
+using System.Text;
+using MassTransit;
+using MassTransit.EntityFrameworkCoreIntegration;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace LeanCode.CQRS.MassTransitRelay.LockProviders;
+
+// Vendoring SqlLockStatementProvider & friends
+// MT implementation uses EF8 incompatible API
+// https://github.com/MassTransit/MassTransit/blob/9e6c78573ad211a70b624fad31382faa331dc4d8/src/Persistence/MassTransit.EntityFrameworkIntegration/EntityFrameworkIntegration/SqlLockStatementProvider.cs
+    public class CustomSqlLockStatementProvider :
```

Comment:
Mkay, _why_?

Reply:
Some random `MethodNotFoundException` :disappointed:

Reply:
So basically we're vendoring MT code to make it work with .NET 8?

Reply:
Well, unless they catch up with this particular thing I don't see any other option.
Those providers are prone to breaking, I remember fixing them in one of the projects some time ago.

Reply:
Mkay, so let's also vendor tests for these (if available).

Reply:
Hmm, there are very limited tests for this, but moving _only_ them would be hard, as there's much orchestration. On the other hand we have at least sql server and sqlite tested in integration tests.

Reply:
Can we also test Postgres one that way (yeah, I'm using this PR as an excuse to add Postgres-specific tests :P )?

Reply:
This would be hard, given that I'd have to copy the whole integration tests. Or somehow run them in theory against two dbs.
I'd rather remove this postgres one as I _need_ this RC1 in one of the projects :stuck_out_tongue: